misc/chrome_extension/panel.js

```diff
@@ -13,138 +16,944 @@
 		}
-		else if(metaType == "metadata"){
+		else if (metaType == 'metadata') {
 			currentJson[0]['metadata'][field] = toAdd;
 		}
 
 		//Add back to page
 		currentJson = JSON.stringify(currentJson, null, 2);
-		textArea.value = currentJson;
+		textAreaElement.value = currentJson;
+		buildVisualFromText();
 	}
 
-	//Hides or shows the field option
-	function toggleField(){
-		let e = document.getElementById("queryToAddMeta");
-		let fieldElement = document.getElementById('fieldToAdd');
-		if(e.value == "metadata"){
-			fieldElement.style.display = 'inline';
+
+	/**
+	 * Hides or shows the field input
+	 * 
+	 */
+	function toggleQuickAddFieldInput() {
+		let queryTypeElement = document.getElementById('queryToAddMeta');
+		let fieldInputElement = document.getElementById('fieldToAdd');
+		if (queryTypeElement.value == 'metadata') {
+			fieldInputElement.style.display = 'inline';
 		}
-		else{
-			fieldElement.style.display = 'none';
+		else {
+			fieldInputElement.style.display = 'none';
 		}
 	}
 
-	//Resets the value table back to default
-	function resetValueTable(){
-		document.getElementById('resultTable').innerHTML = "<tr><th>Field</th><th>Value(s)</th></tr>";
+
+	/**
+	 * Resets the value table back to default
+	 * 
+	 */
+	function resetResultTable() {
+		document.getElementById('resultTable').innerHTML = '<tr><th>Field</th><th>Value(s)</th></tr>';
+	}
+
+	/**
+	 * Turn on the mouseover on the webpage
+	 * Deactivated for now
+	 * 
+	 */
+	function mouseAdd() {
+		chrome.runtime.sendMessage({ tabId: chrome.devtools.inspectedWindow.tabId, mouse: '1' });
 	}
 
-	//The init function
-	function init() {
 
-		//Adds the fetch function to the button
-		document.getElementById('fetch').onclick = fetch;
-		document.getElementById('pretty').onclick = pretty;
-		document.getElementById('queryToAddButton').onclick = addToJson;
-		document.getElementById("queryToAddMeta").onchange = toggleField;
+	/**
+	 * Checks if the key exists in the metadata of the json
+	 * 
+	 * @param {string} key - The key
+	 * @returns {boolean} True if the key exists
+	 */
+	function DoesKeyExist(key) {
+		let textAreaElement = document.getElementById('json-config');
+		let currentValue = textAreaElement.value;
+		let json = JSON.parse(currentValue);
+		return json[0]['metadata'][key] !== undefined
+	}
+
 
-		//Hides or shows the field by default
-		toggleField();
+	/**
+	 * Creates the storage file select
+	 * 
+	 * @param {requestCallback} [callback] - After the select is created and ready to be used
+	 */
+	function initStorageSelect(callback) {
+		try {
+			let select = document.getElementById('storage');
+			select.innerHTML = '<option selected disabled>Select a file to work on</option><option disabled>------------------------------</option>';
+			getStorageValues(function (results) {
+				results.forEach(function (element) {
 
-		//Creates the value table
-		resetValueTable();
+					select.innerHTML += '<option value="' + element + '">' + element + '</option>"';
```

Comment:
JavaScript now has inline templating. Should take advantage of that:
```select.innerHTML += `<option value="${element}">${element}</option>`;```

There was an extra `"` at the end, should not be there.